### DISCUSSION TITLE
PWM DP3265S based 104X52 1/26 scan LED module

### DISCUSSION THREAD

**Discussion by ssevdim** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#693] [General]
Can i use this library for 104x52 1/26 scan rgb led module. I need help.

**Comment by board707** [comment]
I depends on many different things
- type of driver chips on the panel
- its pixel mapping
- your understanding of C++ code and ability to make changes

By default, i.e. "from the box" - the library doesn't support panels with weird width, height or scan factor

**Reply by ssevdim** [reply]
I did everything I can but never found type of driver chips on then panel.  I wrote mail many times to producer. Never reply. The link of the productt is 

https://www.hjcailiang.com/cailiang-outdoor-d3-full-color-smd-led-video-wall-screen-product/

I wil be very happy if you can find any 

Thank you very much  for your reply.

**Reply by board707** [reply]
> I did everything I can but never found type of driver chips on then panel.

Do not ask sellers, usually they doesn't know exact what they sell.
Take a magnifying glass and try your best to read the chip markings on the back side of the panel. There should be at least three types of chips. Write their markings here

**Reply by ssevdim** [reply]
at the end I found the driver chips. : DP3265S  decoding ICN2012.

**Reply by board707** [reply]
> I found the driver chips. : DP3265S

I have a bad news for you - DP3265 is a PWM-type driver that a fully incompatible with the library code

**Reply by vanwkhanhs** [reply]
Have you managed to control it?

**Reply by board707** [reply]
I don't have a DP3265, a work with a DP3264 in progress  but I didn't finished yet.
To see my studies of DP3264 config registers and protocol look at link
https://github.com/hzeller/rpi-rgb-led-matrix/issues/466#issuecomment-2661628621

**Reply by vanwkhanhs** [reply]
It seems that the DP3264 protocol is quite similar to the DP3265 I read about in the datasheet. Let's try our best to complete it!
![image](https://github.com/user-attachments/assets/2407b40b-9462-4d0a-b5f0-d33c75414d82)

**Reply by board707** [reply]
Do you have a datasheet for DP3265 ?

**Reply by vanwkhanhs** [reply]
I already sent it to you

**Reply by vanwkhanhs** [reply]
[Uploading DP3265S_Rev2.3_CN.pdf…]()

**Reply by board707** [reply]
> I already sent it to you

Sorry, I couldn't find the file. If we have communicated before, please provide the link. I'm a little confused about GitHub names.

**Reply by vanwkhanhs** [reply]
http://depuw.com/cn/pro/228.html#min

**Reply by board707** [reply]
@vanwkhanhs 
Thank you very much. I didn't have this datasheet.

**Reply by vanwkhanhs** [reply]
No problem, I hope it can help you.

**Reply by vanwkhanhs** [reply]
Do you know how to calculate gclk_count?

**Reply by vanwkhanhs** [reply]
I'm not sure how many clock cycles WR_CFG and DAT_LAT take

**Reply by board707** [reply]
> how many clock cycles WR_CFG and DAT_LAT take

![Screenshot from 2025-04-09 11-03-06](https://github.com/user-attachments/assets/7ae83de5-59da-4406-82cc-219f582b7afa)

**Reply by vanwkhanhs** [reply]
It’s just that the number of LE pulses equals the number of DCLK pulses.I want to know the total number of pulses supplied for one loop, including the pulses used for data writing.

**Reply by board707** [reply]
For my matrix it is 114 DCLK x  (the number of rows)  - I wrote about this at the link https://github.com/hzeller/rpi-rgb-led-matrix/issues/466#issuecomment-2661628621

**Reply by vanwkhanhs** [reply]
Your data table isn't completely accurate, right? According to the table, there's only one DCLK pulse between VSYNC and PRE_ACT. By the way, how did you calculate the number of DCLK pulses?

**Reply by board707** [reply]
> According to the table, there's only one DCLK pulse between VSYNC and PRE_ACT.

Which table?
On my diagram there are 16 DCLK between VSYNC and PRE_ACT.

**Reply by vanwkhanhs** [reply]
![image](https://github.com/user-attachments/assets/ee7ee894-b5ad-4ecb-9783-7eae1fb1e51f)
Sorry, I meant the diagram. In the datasheet, I only see one DCLK pulse in between.

**Reply by board707** [reply]
I saw 16 clocks on a real matrix. Maybe there can be less, but I haven't tested.

**Reply by vanwkhanhs** [reply]
Do you know the frequency of the signal at the OE pin?

**Reply by board707** [reply]
I didn't quite understand the question. In my program I generate this signal myself.

**Reply by vanwkhanhs** [reply]
I'm not very experienced in this area, but if you happen to have a way to control your LED panel, would you mind sharing it with me? It might really help me figure out how to control mine. I would greatly appreciate your support.

**Reply by board707** [reply]
> if you happen to have a way to control your LED panel, would you mind sharing it with me?

Definitely I will share my code when I will finish it. But now I have only a pulsing diagram that I published on the link above.

**Reply by vanwkhanhs** [reply]
Great. I’ve followed you now, see you again soon!

**Reply by Mert-Taner** [reply]
I have been trying to run it with esp32 for 2 months but I could not find a solution.

**Comment by Mert-Taner** [comment]
Hello, how did you drive the dp3265s chip led panel with esp32? Can you help me? I've been working on it for about 2 months.
